src/app/resources/iis_scripts/local_http.ps1

```diff
@@ -0,0 +1,59 @@
+param(
+    [string]
+    $requestBase64
+)
+
+$ErrorActionPreference = "Stop"
+$contentEncoding = [System.Text.Encoding]::UTF8
+
+function stringify($content) {
+    if ((!$content) -or ($content -is [System.String])) {
+        return $content
+    } elseif ($content -is [System.Byte[]]) {
+        return [System.Convert]::ToBase64String($content)
+    } else {
+        return $content.ToString()
+    }
+}
+
+$decoded = $contentEncoding.GetString([System.Convert]::FromBase64String($requestBase64))
+$reqObj = ConvertFrom-Json $decoded
+$uri = [System.UriBuilder]$reqObj.url
+$uri.Host = "localhost"
+$req = @{ "Uri" = $uri.ToString() }
+
+if ($reqObj.method) {
+    $req.Method = [Microsoft.PowerShell.Commands.WebRequestMethod]$reqObj.method
+}
+
+if ($reqObj._body) {
+    $req.Body = $reqObj._body
+}
+
+if ($reqObj.headers) {
+    $req.Headers = @{}
+    foreach ($prop in $reqObj.headers | Get-Member -MemberType NoteProperty | Select-Object -ExpandProperty Name) {
+        ## NOTE: not all http headers follow the same syntax. This delimitation logic is assumed to work for all headers we use
+        $req.Headers[$prop] = $reqObj.headers.$prop -Join ","
+    }
+}
+
+try {
+    $res = Invoke-WebRequest -UseBasicParsing -UseDefaultCredentials @req
+} catch {
+    $res = $_.Exception.Response
+} finally {
+    if (!$res) {
+        throw "No response is returned."

```

Comment:
When exception happens, I think we should pass the error code and the url information instead of returning "No response is returned." so that users can troubleshoot.  As far as I remember, invoke-webrequest throw exception for permisson error or 5xx error.

Reply:
I did assign the response object in the catch block though. The !$res scenario should never happen since non-success status code would be captured. In fact I am using the error status code on my connection logic.
This line is only here so there wouldn't be any NPE but if you do reach this line it's a completely unexpected error. Not HTTP error

Reply:
The catch statement won't work. Please run below commands and you can see the $res is still $null value. 

```
$res = $null; 
try { 
    invoke-webrequest http://localhost:1234/bogus;
} 
catch { 
    $res = $_.Exception.Response 
}; 
$res
```

Reply:
The status code 4XX and 5XX should already work. In fact I think all the non zero status code scenario works.
However, I think this bring up a good point. I have modified the script to return the error message when status code is 0 in this revision